src/cdk/overlay/overlay-ref.ts

```diff
@@ -79,6 +86,18 @@ export class OverlayRef implements PortalOutlet {
     }
 
     this._positionStrategy = _config.positionStrategy;
+
+    // Users could open the overlay from an `effect`, in which case we need to
+    // run the `afterRender` as `untracked`. We don't recommend that users do
+    // this, but we also don't want to break users who are doing it.
+    this._afterRenderRef = untracked(() =>
+      afterRender(
```

Comment:
Would there be any perf implications given that we'd be running this constantly for the lifetime of the overlay?

Reply:
We're not actually doing any work, other than just nexting an observable, but I guess its a fair question how much impact even just queuing the `afterRender` has. @alxhub @atscott do either of you know if we should be concerned about the performance here?

Reply:
I don't see how this would be much different from a subscription to `onStable`. `afterRender` executes slightly more often but I can't imagine this would have any performance implications.